sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java

```diff
@@ -311,10 +307,18 @@ private URL getBuildEndpoint(ConfigurationClientCredentials buildCredentials) {
         if (endpoint != null) {
             return endpoint;
         } else if (buildCredentials != null) {
-            return buildCredentials.getBaseUri();
+            return getURLFromString(buildCredentials.getAppConfigurationURL());
         } else {
             return null;
         }
     }
+
+    private URL getURLFromString(String endpoint) {
```

Comment:
Instead of adding this you should also change the constructor for `ConfigurationAsyncClient` to take a `String` instead of URL, all it is doing is calling `toString` on the passed `URL`. Kind of pointless to add all this exception wrapping and conversion handling.